hack/choco/update-choco-package.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+ 
+# Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+ 
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+ 
+version="${1:?TCE version argument empty. Example usage: ./hack/choco/update-choco-package.sh v0.10.0}"
+: "${GITHUB_TOKEN:?GITHUB_TOKEN is not set}"
+ 
+ 
+temp_dir=$(mktemp -d)
+ 
+pushd "${temp_dir}"
+
+TCE_REPO="https://github.com/vmware-tanzu/community-edition/" 
+TCE_REPO_RELEASES_URL="https://github.com/vmware-tanzu/community-edition/releases"
+TCE_WINDOWS_TAR_BALL_FILE="tce-darwin-amd64-${version}.tar.gz"
+TCE_CHECKSUMS_FILE="tce-checksums.txt"
+ 
+echo "Checking if the necessary files exist for the TCE ${version} release"
+ 
+curl -f -I -L \
+   "${TCE_REPO_RELEASES_URL}/download/${version}/${TCE_WINDOWS_TAR_BALL_FILE}" > /dev/null || {
+       echo "${TCE_WINDOWS_TAR_BALL_FILE} is not accessible in TCE ${version} release"
+       exit 1
+   }
+ 
+wget "${TCE_REPO_RELEASES_URL}/download/${version}/${TCE_CHECKSUMS_FILE}" || {
+   echo "${TCE_CHECKSUMS_FILE} is not accessible in TCE ${version} release"
+   exit 1
+}
+ 
+git clone "${TCE_REPO}"
+
+cd community-edition
+
+# make sure we are on main branch before checking out
+git checkout main
+ 
+PR_BRANCH="update-tce-to-${version}-${RANDOM}"
+ 
+# Random number in branch name in case there's already some branch for the version update,
+# though there shouldn't be one. There could be one if the other branch's PR tests failed and didn't merge
+git checkout -b "${PR_BRANCH}"
+ 
+# Replacing old version with the latest stable released version.
+# Using -i so that it works on Mac and Linux OS, so that it's useful for local development.
+sed -i -e "s/\(\$releaseVersion =\).*/\$releaseVersion = ""'${version}'""/g" hack/choco/tools/chocolateyinstall.ps1
```

Comment:
If we know this is always going to be running on our Linux GitHub Action environment, the `-i` can be dropped and you don't need to do the `rm` in the next line.

Otherwise, you could do detection like we do in https://github.com/vmware-tanzu/community-edition/blob/5b8537e2c64b902553b57956af0ba650b18b2387/hack/packages/create-package.sh#L27

Reply:
Hello @stmcginnis I was trying the above code change and found that for linux too we need `-i` flag to change the same file so as for both OS mac/linux we need this then there is no need for detection. Please correct me if i interpret wrong yours above suggestion.